### PR TITLE
Add a user route to lookup assetstores by name

### PR DESCRIPTION
**Pull request by alesgenova**
Admins are the only ones allowed to fetch the list of assetstores.
When creating an assetstore for a cluster, if another process has
created it there may be a need to lookup the assetstore as a regular
user. This additional rout allows for it.

### CODE REVIEW

File: girder/cumulus/cumulus_plugin/assetstore.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from girder.api import access
+from girder.api.describe import Description, autoDescribeRoute
+from girder.models.assetstore import Assetstore
+
+@access.user
```
Comment (psavery) [inline girder/cumulus/cumulus_plugin/assetstore.py]: We will need to add the appropriate `TokenScope` here (see #359), or else this endpoint will not be accessible to user api keys for which "Allow all actions on behalf of my user" was not checked (as is done in mongochemclient).

Comment (alesgenova) [reply girder/cumulus/cumulus_plugin/assetstore.py]: Thank you for the heads up, I fixed it

**Comment by cjh1** [comment]
@psavery Can you please check that this resolves the issue you where seeing. Thanks!

**Comment by psavery** [comment]
Unfortunately, it looks like I am still getting the same error (pasted below).

```
celery_command_1  | Traceback (most recent call last):
celery_command_1  |   File "/usr/local/lib/python3.6/site-packages/celery/app/trace.py", line 240, in trace_task
celery_command_1  |     R = retval = fun(*args, **kwargs)
celery_command_1  |   File "/usr/local/lib/python3.6/site-packages/celery/app/trace.py", line 438, in __protected_call__
celery_command_1  |     return self.run(*args, **kwargs)
celery_command_1  |   File "/cumulus/cumulus/taskflow/__init__.py", line 117, in wrapped
celery_command_1  |     return func(celery_task, *args, **kwargs)
celery_command_1  |   File "/openchemistrypy/taskflows/taskflows/__init__.py", line 287, in postprocess_description
celery_command_1  |     upload_job_output_to_folder(cluster, description_job, girder_token=task.taskflow.girder_token)
celery_command_1  |   File "/cumulus/cumulus/tasks/job.py", line 778, in upload_job_output_to_folder
celery_command_1  |     assetstore_id = get_assetstore_id(girder_token, cluster)
celery_command_1  |   File "/cumulus/cumulus/transport/files/__init__.py", line 78, in get_assetstore_id
celery_command_1  |     check_status(r)
celery_command_1  |   File "/cumulus/cumulus/common/__init__.py", line 32, in check_status
celery_command_1  |     request.raise_for_status()
celery_command_1  |   File "/usr/local/lib/python3.6/site-packages/requests/models.py", line 940, in raise_for_status
celery_command_1  |     raise HTTPError(http_error_msg, response=self)
celery_command_1  | requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: http://girder:8080/api/v1/assetstore?limit=9223372036854775807
```

**Comment by psavery** [comment]
Maybe my cumulus isn't updated for some reason. I know that endpoint should have changed. Let me have a look...

**Comment by psavery** [comment]
Yep! I think this fixes the issue!

It looks like there are multiple cumulus instances in our dev stack. I just had to make sure the right one was using this branch (it was one of the celery workers, and I had to restart it after updating its' cumulus).

I see from the output that the new rest endpoint was called, and all the jobs succeeded!

**Pull request merged**